behavior_tree_core/CMakeLists.txt

```diff
@@ -21,101 +20,141 @@ add_action_files(
   FILES BT.action
 )
 
-
 generate_messages(
-  DEPENDENCIES actionlib_msgs std_msgs # Or other packages containing msgs
+  DEPENDENCIES
+    actionlib_msgs
+    std_msgs
 )
 
-
-add_definitions(-Wall -lglut -lGL -std=c++11)
+add_definitions(-std=c++11)
 
 catkin_package(
   CATKIN_DEPENDS
     actionlib_msgs
     behavior_tree_msgs
+    message_runtime
+    std_msgs
   INCLUDE_DIRS include
-  LIBRARIES ${PROJECT_NAME}
+  LIBRARIES ${PROJECT_NAME}_lib
 )
 
 roslint_cpp()
 
-#########################################################
-# FIND GLUT
-#########################################################
-find_package(GLUT REQUIRED)
-include_directories(${GLUT_INCLUDE_DIRS})
-link_directories(${GLUT_LIBRARY_DIRS})
-add_definitions(${GLUT_DEFINITIONS})
-if(NOT GLUT_FOUND)
-    message(ERROR " GLUT not found!")
-endif(NOT GLUT_FOUND)
-
-#########################################################
-# FIND OPENGL
-#########################################################
-find_package(OpenGL REQUIRED)
-include_directories(${OpenGL_INCLUDE_DIRS})
-link_directories(${OpenGL_LIBRARY_DIRS})
-add_definitions(${OpenGL_DEFINITIONS})
-if(NOT OPENGL_FOUND)
-    message(ERROR " OPENGL not found!")
-endif(NOT OPENGL_FOUND)
-
-INCLUDE_DIRECTORIES(${catkin_INCLUDE_DIRS} include)
+file(GLOB_RECURSE BTHeadLibrary include/*.h
+  EXCLUDE
+    include/draw.h
+)
 
-file(GLOB_RECURSE BTHeadLibrary include/*.h)
+list(SORT BTHeadLibrary)
 
 set(BTSrcLibrary
 src/action_node.cpp
+src/actions/action_test_node.cpp
+src/actions/ros_action.cpp
 src/behavior_tree.cpp
+src/blackboard/blackboard.cpp
 src/condition_node.cpp
+src/conditions/condition_test_node.cpp
+src/conditions/ros_condition.cpp
 src/control_node.cpp
-src/draw.cpp
+src/dot_bt.cpp
 src/exceptions.cpp
+src/fallback_node.cpp
+src/fallback_node_with_memory.cpp
 src/leaf_node.cpp
-src/tick_engine.cpp
 src/parallel_node.cpp
-src/fallback_node.cpp
 src/sequence_node.cpp
-src/fallback_node_with_memory.cpp
 src/sequence_node_with_memory.cpp
+src/tick_engine.cpp
 src/tree_node.cpp
-src/actions/action_test_node.cpp
-src/conditions/condition_test_node.cpp
-src/actions/ros_action.cpp
-src/conditions/ros_condition.cpp
-src/blackboard/blackboard.cpp
-src/dot_bt.cpp
 )
 
+#########################################################
+# FIND GLUT AND OPENGL
+#########################################################
+find_package(GLUT)
+if(NOT GLUT_FOUND)
+    message("GLUT not found!")
+endif()
+
+find_package(OpenGL)
+if(NOT OPENGL_FOUND)
+    message("OPENGL not found!")
+endif()
+
+if(GLUT_FOUND AND OPENGL_FOUND)
+  include_directories(${GLUT_INCLUDE_DIRS})
+  add_definitions(${GLUT_DEFINITIONS})
+  
+  include_directories(${OpenGL_INCLUDE_DIRS})
```

Comment:
I may be wrong, but on my machine it appears the correct variable is `OPENGL_INCLUDE_DIR`.